clang/unittests/Driver/ToolChainTest.cpp

```diff
@@ -316,6 +318,52 @@ TEST(ToolChainTest, VFSSolarisMultiGCCInstallation) {
   }
 }
 
+MATCHER_P(jobHasArgs, Substr, "") {
+  const driver::Command &C = arg;
+  std::string Args = "";
+  llvm::ListSeparator Sep(" ");
+  for (const char *Arg : C.getArguments()) {
+    Args += Sep;
+    Args += Arg;
+  }
+  if (is_style_windows(llvm::sys::path::Style::native))
+    std::replace(Args.begin(), Args.end(), '\\', '/');
+  if (llvm::StringRef(Args).contains(Substr))
+    return true;
+  *result_listener << "whose args are '" << Args << "'";
```

Comment:
Not familiar with the unit testing set up ... do you need the "whose args are" bit?

Reply:
Yeah, this is printed only when the matcher fails. It's needed for a decent error message as the job itself has no useful printer.

Reply:
Cool, sounds good.